platform/admin/ui/src/main/webapp/js/views/Alerts.view.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global define*/
+define([
+        'marionette',
+        'text!templates/alerts.handlebars',
+        'icanhaz'
+    ],
+    function (Marionette, alertsTemplate, ich) {
+
+        ich.addTemplate('alertsTemplate', alertsTemplate);
+
+        var AlertsView = {};
+
+        AlertsView.View = Marionette.ItemView.extend({
+            template: 'alertsTemplate',
+            tagName: 'table',
+            events: {
+                'shown.bs.collapse': 'toggleDetailsMsg',
+                'hidden.bs.collapse': 'toggleDetailsMsg'
+            },
+            modelEvents: {
+                'change': 'render'
+            },
+            /*jshint -W030 */
+            toggleDetailsMsg: function () {
+                var model = this.model;
+                model.get('button') === 'Show' ? model.set('button', 'Hide') : model.set('button', 'Show');
+                model.get('collapse') === 'out' ? model.set('collapse', 'in') : model.set('collapse', 'out');
+            },
+            serializeData: function () {
+                var json = this.model.toJSON();
+                json.collapseId = parseInt((Math.random() * 10000), 10);
```

Comment:
The chance of a collision still seems high.  For example, it would be around 11% if there were 50 hidden alert modals.  I would suggest using something like `Math.random() * Math.pow(2, 32)`.

http://www.wolframalpha.com/input/?i=1-e%5E%28%28-50*%2850-1%29%29%2F%282*10000%29%29
http://preshing.com/20110504/hash-collision-probabilities/